### PR TITLE
[MM-47606]: Blank message in threads if send leave/join message preference is off

### PR DESCRIPTION
**Pull request by AshishDhama**
#### Summary
Blank message in threads if send leave/join message preference is off.

#### Ticket Link
Fixes https://mattermost.atlassian.net/jira/software/c/projects/MM/boards/91?modal=detail&selectedIssue=MM-47606

#### Related Pull Requests
NA

#### Screenshots
|  before  |  after  |
|----|----|
| image-20221014-195045 | Screenshot 2022-11-14 at 1 27 44 PM

#### Release Note
```release-note
* Blank message in threads if send leave/join message preference is off
```

### CODE REVIEW

File: packages/mattermost-redux/src/selectors/entities/posts.ts [hunk 1]
```diff
@@ -351,19 +353,26 @@ export function makeGetPostsForThread(): (state: GlobalState, rootId: string) =>
     return createIdsSelector(
         'makeGetPostsForThread',
         getAllPosts,
+        getCurrentUser,
         (state: GlobalState, rootId: string) => state.entities.posts.postsInThread[rootId],
         (state: GlobalState, rootId: string) => state.entities.posts.posts[rootId],
-        (posts, postsForThread, rootPost) => {
+        getMyPreferences,
```
Comment (koox00) [inline packages/mattermost-redux/src/selectors/entities/posts.ts:359]: FYI we have a selector that returns true/false based on preferences [`shouldShowJoinLeaveMessages`](https://github.com/mattermost/mattermost-webapp/blob/master/packages/mattermost-redux/src/utils/post_list.ts#L28)

Comment (AshishDhama) [reply packages/mattermost-redux/src/selectors/entities/posts.ts:361]: Done thanks

Comment (koox00) [reply packages/mattermost-redux/src/selectors/entities/posts.ts:361]: `getMyPreferences` and `myPreferences` should not be needed now, right? how come eslint didn't complain on `myPreferences` not being used 🤔 , am I missing something?

File: components/threading/global_threads/thread_pane/thread_pane.test.tsx [hunk 2]
```diff
@@ -13,6 +13,9 @@ import Header from 'components/widgets/header';
 import FollowButton from 'components/threading/common/follow_button';
 import Button from 'components/threading/common/button';
 
+import TestHelper from '../../../../packages/mattermost-redux/test/test_helper';
+import {UserProfile} from '@mattermost/types/users';
```
Comment (M-ZubairAhmed) [inline components/threading/global_threads/thread_pane/thread_pane.test.tsx:17]: minor nit : import rearrangments

File: components/threading/global_threads/thread_pane/thread_pane.test.tsx [hunk 3]
```diff
@@ -85,6 +92,7 @@ describe('components/threading/global_threads/thread_pane', () => {
                     },
                 },
                 users: {
+                    profiles,
```
Comment (M-ZubairAhmed) [inline components/threading/global_threads/thread_pane/thread_pane.test.tsx:95]: would you be open in writing a unit test which return shouldFilterJoinLeavePost true or false?

**Comment by M-ZubairAhmed** [comment]
removed harrison, since he is PTO

File: components/threading/global_threads/thread_pane/thread_pane.test.tsx (hunk 2 above)
Comment (AshishDhama) [reply components/threading/global_threads/thread_pane/thread_pane.test.tsx:17]: Is it really important, why don’t we add this in eslint.

**Review by M-ZubairAhmed — APPROVED** [review]

**Review by koox00 — COMMENTED** [review]
👍

**Review by koox00 — APPROVED** [review]

**Comment by jgilliam17** [comment]
Thanks @AshishDhama 
Tested, looks good to merge.
Verified when leave/join message setting is off, blank message in threads in no longer shown. 
[E2E report](https://automation-dashboard.vercel.app/cycles/09ffa2be-2c7f-46ba-8d1c-9f17e317699a) - no PR related failures

**Review by jgilliam17 — APPROVED** [review]

**Pull request merged**